### ISSUE TITLE
Unable to use AccountsMgmt/v1/AccessToken client

### ISSUE THREAD

**Issue by tiwillia** [openshift-online/ocm-sdk-go#37]
There are two issues blocking use of the AccessToken client:

1. It is not possible, through the client, to set and send an empty request body. Attempting to do so results in the following error:
```
client := connection.AccountsMgmt().V1().AccessToken()
response, err := client.Post().Send()

request body is mandatory for the 'POST' method
```

2. The response body for AccessToken cannot be defined in ocm-api-model using the current structure. The response body looks like this, with an indeterminate number of `auths` values:
```
{
  "auths": {
    "cloud.openshift.com": {
      "auth": "REDACTED"
      "email": "tiwillia@redhat.com"
    },
    "quay.io": {
      "auth": "REDACTED"
      "email": "tiwillia@redhat.com"
    },
    "registry.connect.redhat.com": {
      "auth": "REDACTED"
      "email": "tiwillia@redhat.com"
    },
    "registry.redhat.io": {
      "auth": "REDACTED"
      "email": "tiwillia@redhat.com"
    }
  }
}
```

We need some way to properly model this response so that the SDK and parse the JSON and return it as `map[string]interface{}` for consumption.

**Comment by tiwillia** [comment]
The workaround today is to use the connection `Path()` function to establish the empty post request without using the `AccessToken` client:
```
connection.Post("/api/accounts_mgmt/v1/access_token").Body([]byte{}).Send()
```

**Comment by jhernand** [comment]
The problem with the empty body has been solved in https://github.com/openshift-online/ocm-sdk-go/pull/38.

**Comment by jhernand** [comment]
The support for maps in the code generator has been improved in https://github.com/openshift-online/ocm-api-metamodel/pull/21. That should allow you to define the following types:

```
struct AccessToken {
       Auths [string]AccessTokenAuth
}

struct AccessTokenAuth {
        Username string
        Email string
}
```

I think that is sufficient to solve the second issue that you mention. The pull request for the model should now be similar to this: https://github.com/openshift-online/ocm-api-model/pull/33

Once that is merged the SDK can be updated to use it: https://github.com/openshift-online/ocm-sdk-go/pull/40

Then you will be able to get access tokens with the SDK as shown in this example: https://github.com/jhernand/ocm-sdk-go/blob/add_access_token_auth_type/examples/get_access_token.go

@tiwillia Is that enough for what you need? If so I will update and merge those pull requests.

**Comment by jhernand** [comment]
This is fixed in release 0.1.44. See #41.
